tests/infrastructure/taints-and-tolerations.go

```diff
@@ -165,9 +98,141 @@ var _ = DescribeInfra("[rfe_id:4126][crit:medium][vendor:cnv-qe@redhat.com][leve
 			selectedNode, err = virtClient.CoreV1().Nodes().Patch(context.Background(), selectedNode.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			Consistently(terminatedPodsCn, 5*time.Second).ShouldNot(Receive(), "pods should not terminate")
-			stopCn <- true
+			taintedNodeName = nodeName
```

Comment:
I think this test can still be unstable, since if at this point the taint has succeeded but something else has gone wrong, the taint will not get removed in the `AfterSuite`. Better set the `taintedNodeName` directly after line 87, and check in the `AfterEach` whether the taint is present on the node.

And if you follow the suggestion, I'd rename `taintedNodeName` into `possiblyTaintedNodeName`.

WDYT?

Reply:
Good idea, I will change it.

Reply:
Do you mean that there can be an edge case where the `Patch()` on line 98 returns an error, but the taint succeeded?

Reply:
done